python/KMS_KEYS_TO_NOT_DELETE/KMS_KEYS_TO_NOT_DELETE.py

```diff
@@ -0,0 +1,455 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+'''
+#####################################
+##            Gherkin              ##
+#####################################
+
+Rule Name:
+	KMS_KEYS_TO_NOT_DELETE
+
+Description:
+	Check that Customer Managed keys are not scheduled for deletion. The rule is NON_COMPLAINT if the Customer Managed Keys are scheduled for deletion. This rule does not include AWS-managed and Imported Keys.
+
+Trigger:
+	Periodic
+
+Reports on:
+	AWS::KMS::Key
+
+Rule Parameters:
+	kmsKeyIds (Optional)
+	Comma-separated list of specific Customer Managed Key Ids, that are expected not to be scheduled for deletion.
+
+Scenarios:
+    	Scenario 1:
+    	Given: At least 1 CMK is present in the AWS Account
+      	  And: Rule parameter kmsKeyIds are configured and not valid
+     	 Then: Return ERROR
+
+    	Scenario 2:
+    	Given: No CMKs present
+         Then: Return NOT_APPLICABLE
+
+    	Scenario 3:

```

Comment:
Can you move this scenario just before the existing Scenario 6. It will be easier to read the Gherkin.